Applications/spire/source/charting/chart_view.cpp

```diff
@@ -127,7 +127,6 @@ void ChartView::set_region(const ChartPoint& top_left,
     m_bottom_right.m_x);
   m_candlestick_promise.then([&] (auto result) {
     m_candlesticks = std::move(result.Get());
-    update();
     if(m_is_auto_scaled) {
```

Comment:
This is still a bug, you must guarantee only one update is performed. The problem with the old implementation was that you could potentially be updating twice, now it's possible you never update.